.github/workflows/check-uncommitted.yml

```diff
@@ -31,5 +32,7 @@ jobs:
       - name: Rebuild the dist/ directory
         run: npm run build
 
-      - name: Compare the expected and actual dist/ directories
-        run: bin/check-diff
```

Comment:
As I recall, the intent of using this script was to have the Check Run show the diff on failure so you didn't need to run it on your machine to see what the issue was.

This was useful as we had environment-specific issues early on where the Typescript compile wasn't properly encapsulated and produced a diff on Actions, but none locally.

> It also checks for any delta in the git repo, not just the dist/ directory. Any change should fail CI until it's either committed or added to .gitignore.

That's fair - I don't have a strong opinion on this, I think it was originally just the dist/ directory to keep the CI build focused on verifying the distributed code as any other changes were less impactful.

Reply:
This is useful, so I kept it, and then augmented with another job that looks for any other uncommitted files.